components/search/src/selector.rs

```diff
@@ -317,4 +356,262 @@ mod tests {
+                  "identifier": "test3",
+                  "base": {
+                    "name": "Test 3",
+                    "classification": "general",
+                    "urls": {
+                      "search": {
+                        "base": "https://example.com/3",
+                        "method": "GET",
+                        "searchTermParamName": "trek"
+                      }
+                    }
+                  },
+                  "variants": [{
+                    "environment": {
+                      "distributions": ["starship"]
+                    }
+                  }],
+                },
+                {
+                  "recordType": "defaultEngines",
+                  "globalDefault": "test1",
+                }
+              ]
+            })
+            .to_string(),
+        );
+        assert!(
+            config_result.is_ok(),
+            "Should have set the configuration successfully. {:?}",
+            config_result
+        );
+
+        let mut result = Arc::clone(&selector).filter_engine_configuration(SearchUserEnvironment {
+            locale: "fi".into(),
+            region: "FR".into(),
+            update_channel: SearchUpdateChannel::Default,
+            distribution_id: String::new(),
+            experiment: String::new(),
+            app_name: SearchApplicationName::Firefox,
+            version: String::new(),
+        });
+
+        assert!(
+            result.is_ok(),
+            "Should have filtered the configuration without error. {:?}",
+            result
+        );
+        assert_eq!(
+            result.unwrap(),
+            RefinedSearchConfig {
+                engines: vec!(
+                    SearchEngineDefinition {
+                        aliases: Vec::new(),
+                        charset: "UTF-8".to_string(),
+                        classification: SearchEngineClassification::General,
+                        identifier: "test1".to_string(),
+                        name: "Test 1".to_string(),
+                        order_hint: None,
+                        partner_code: String::new(),
+                        telemetry_suffix: None,
+                        urls: SearchEngineUrls {
+                            search: SearchEngineUrl {
+                                base: "https://example.com/1".to_string(),
+                                method: "GET".to_string(),
+                                params: Vec::new(),
+                                search_term_param_name: Some("q".to_string())
+                            },
+                            suggestions: None,
+                            trending: None
+                        }
+                    },
+                ),
+                app_default_engine_id: "test1".to_string(),
+                app_default_private_engine_id: None
+            }, "Should have selected the single engine, as the environments do not match for the other two"
```

Comment:
I might have missed this one in my earlier review. But I think this can be revised to more specific wording: 

"Should have selected test1 for all matching locales, as the environments do not match for the other two"